### ISSUE TITLE
Support optional ruleset metadata attributes in violation analysis reports

### ISSUE THREAD

**Issue by tsanders-rh** [konveyor/analyzer-lsp#1048]
### Background
The Konveyor community is adding optional metadata attributes to the ruleset schema to improve rule classification and provide better context for migration analysis:
- [Enhancement PR #255](https://github.com/konveyor/enhancements/pull/255) - Adding `migration_complexity` field
- [Enhancement Issue #254](https://github.com/konveyor/enhancements/issues/254) - Adding `domain`, `category`, `metadata` (severity, confidence, migrationPath), and other optional fields

These attributes are designed to be completely optional and backward-compatible, but they would be valuable to include in violation analysis reports for better decision-making and routing (e.g., AI automation thresholds based on complexity).

### Current State
Currently, analyzer-lsp only extracts and passes through a fixed set of rule metadata fields:
- `ruleID`
- `description` 
- `category` (Potential/Mandatory/Optional)
- `labels`
- `effort`
- `links`
- `customVariables`

**Any additional optional fields in the ruleset YAML are discarded during parsing** and not included in the violation analysis report.

The `konveyor.Violation` struct has an `Extras json.RawMessage` field (violations.go:97) that is reserved for additional data, but is currently always set to an empty byte array (engine.go:740).

### Proposed Solution

Use the existing `Extras` field to pass through all optional/unknown rule metadata. This approach is simpler and more flexible than adding explicit struct fields.

#### 1. Update Rule Metadata Structure
Add a simple extras map to `engine.RuleMeta` (engine/conditions.go:108-115):
```go
type RuleMeta struct {
    RuleID      string
    Description string
    Category    *konveyor.Category
    Labels      []string
    Effort      *int
    UsesHasTags bool
    Extras      map[string]interface{}  // All optional/unknown fields
}
```

#### 2. Update Parser
Extend `parser.addRuleFields()` (parser/rule_parser.go:517-585) to collect any fields that aren't part of the core schema:
```go
func (r *RuleParser) addRuleFields(rule *engine.Rule, ruleMap map[string]interface{}) {
    // ... existing field extraction ...
    
    // Collect any extra fields not in the core schema
    knownFields := map[string]bool{
        "ruleID": true, "when": true, "message": true, "tag": true,
        "description": true, "category": true, "labels": true,
        "effort": true, "links": true, "customVariables": true,
    }
    
    extras := make(map[string]interface{})
    for key, value := range ruleMap {
        if !knownFields[key] {
            extras[key] = value
        }
    }
    
    if len(extras) > 0 {
        rule.Extras = extras
    }
}
```

#### 3. Populate Violation Extras
Update violation construction in `engine.go:735` to marshal and include the extras:
```go
extrasJSON := []byte{}
if len(rule.Extras) > 0 {
    if data, err := json.Marshal(rule.Extras); err == nil {
        extrasJSON = data
    }
}

return konveyor.Violation{
    Description: rule.Description,
    Labels:      rule.Labels,
    Category:    rule.Category,
    Incidents:   incidents,
    Extras:      extrasJSON,  // Now populated with optional fields
    Effort:      rule.Effort,
    Links:       rule.Perform.Message.Links,
}, nil
```

### Benefits
- **Flexible**: Any new optional fields automatically flow through without code changes
- **Simple**: Minimal changes, uses existing infrastructure
- **Backward compatible**: Extras remain empty for rules without optional fields
- **Type-safe consumption**: Consumers can unmarshal specific fields they care about
- **Future-proof**: Supports the evolving ruleset schema without constant updates

### Example Input/Output

**Input ruleset:**
```yaml
- ruleID: javax-to-jakarta-00001
  message: "Replace javax imports with jakarta"
  category: mandatory
  effort: 1
  migration_complexity: trivial
  domain: "jakarta-migration"
  metadata:
    severity: "high"
    confidence: "high"
```

**Output violation:**
```yaml
violations:
  javax-to-jakarta-00001:
    description: "Replace javax imports with jakarta"
    category: mandatory
    effort: 1
    extras:
      migration_complexity: trivial
      domain: "jakarta-migration"
      metadata:
        severity: "high"
        confidence: "high"
    incidents: [...]
```

### Related Files
- `engine/conditions.go:108-115` - RuleMeta structure
- `parser/rule_parser.go:517-585` - addRuleFields() function
- `engine/engine.go:735-743` - Violation construction
- `output/v1/konveyor/violations.go:97` - Violation.Extras field definition
